src/qibolab/platforms/platform.py

```diff
@@ -45,15 +45,23 @@ def sweep(self, sequence, *sweepers, nshots=1024, relaxation_time=None, average=
         )
 
     def set_lo_drive_frequency(self, qubit, freq):
+        if self.qubits[qubit].drive.local_oscillator is None:
+            raise_error(NotImplementedError, f"Qubit {qubit} drive does not have a local oscillator")
```

Comment:
Another way to implement this error to avoid repeating in all functions here is to make `local_oscillator` a `@property` of the `Channel` object and raise the error in the getter and setter.

Reply:
I tried to implement this, but I'm not sure I completely understood the idea.
Also, some other tests fail in this way so maybe it's better to temporary leave the repeated errors as before (?)

Reply:
It looks good to me. I think you just need to update the failing test from
```py
assert channel.local_oscillator is None
```
to
```py
with pytest.raises(NotImplementedError):
    _ = channel.local_oscillator
```
in order to test the error.